src/user-attribute-handler.js

```diff
@@ -13,12 +13,22 @@ function UserAttributeHandler(common) {
 UserAttributeHandler.prototype.onRemoveUserAttribute = function(
     key,
     mParticleUser
-) {};
+) {
+    delete this.common.userAttributes[key];
+    window.heap.addUserProperties(this.common.userAttributes);
+};
 UserAttributeHandler.prototype.onSetUserAttribute = function(
     key,
     value,
     mParticleUser
-) {};
+) {
+    if (!this.common.userAttributes) {
```

Comment:
What would happen in this case if the `mParticleUser` already has attributes? Wouldn't we want those included as well?

Reply:
Good point. I updated this to assign the common.userAttributes to the return from mParticleUser.getAllUserAttributes.

Reply:
@alexs-mparticle - Interestingly, this change has caused some of my unit tests to fail. Looking into the specifics of why, and it seems like the generated `test-bundle.js` is not passing `mParticleUse`r to `onSetUserAttribute`, but is instead passing `forwarderSettings`. The same is true of `onRemoveUserAttribute`. 
<img width="538" alt="Screenshot 2024-06-21 at 10 47 19 AM" src="https://github.com/mparticle-integrations/mparticle-javascript-integration-heap/assets/100377552/c84cdcb0-5248-47bc-ad91-552aa70de67e">

I'm not sure how to resolve this, do you have any idea on if this is expected or if its something I can configure/change in this repo?

Reply:
This is the function in the user-attribute-handler.js from the skeleton. 
<img width="524" alt="Screenshot 2024-06-21 at 11 09 27 AM" src="https://github.com/mparticle-integrations/mparticle-javascript-integration-heap/assets/100377552/8499cf6f-5eaa-4706-8d82-b52d296c8b94">

Reply:
I think we can probably chalk this up to me over thinking things and forgetting how some parts of the Web SDK work.

In the [core SDK](https://github.com/mParticle/mparticle-web-sdk/blob/33efc258f37e7ed121e4b56249407d50d397bfa1/src/identity.js#L933), we only pass `key` and `value`, which is consistent with what you're doing here.

We also do the [same as what you're doing in the GA4 Kit](https://github.com/mparticle-integrations/mparticle-javascript-integration-google-analytics-4/blob/master/packages/GA4Client/src/user-attribute-handler.js#L25).

If you wanted to check the actual user attributes for `mParticleUser`, you'd have to call the user from the core SDK like so:

```javascript
window.mParticle.getInstance().Identities.getCurrentUser().getAllUserAttributes()
```

However I think my initial assumption was wrong so feel free to ignore this suggestion.